modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_zephyr.c

```diff
@@ -201,3 +256,108 @@ uint64_t nrf_802154_platform_sl_lptimer_sync_lpticks_get(void)
 {
 	return m_sync_timer.target_time;
 }
+
+nrf_802154_sl_lptimer_platform_result_t nrf_802154_platform_sl_lptimer_hw_task_prepare(
+	uint64_t fire_lpticks,
+	uint32_t ppi_channel)
+{
+	bool done_on_time;
+	nrf_802154_sl_mcu_critical_state_t mcu_cs_state;
+
+	if (!hw_task_state_set(HW_TASK_STATE_IDLE, HW_TASK_STATE_SETTING_UP)) {
+		/* The only one available set of peripherals is already used. */
+		return NRF_802154_SL_LPTIMER_PLATFORM_NO_RESOURCES;
+	}
+
+	/* Allocate the channel if not already done. */
+	if (m_hw_task.chan == RTC_CHAN_INVALID) {
+		/* The channel allocation cannot take place during module
+		 * initialization, because for nRF53 it would run
+		 * out of resources - some other module temporarily needs
+		 * an rtc channel to initialize. For this reason, this is where
+		 * the allocation is made.
+		 * Once a channel has been successfully allocated, it will be held
+		 * until nrf_802154_platform_sl_lp_timer_deinit is called.
+		 */
+		m_hw_task.chan = z_nrf_rtc_timer_chan_alloc();
+		if (m_hw_task.chan < 0) {
+			m_hw_task.chan = RTC_CHAN_INVALID;
+			hw_task_state_set(HW_TASK_STATE_SETTING_UP, HW_TASK_STATE_IDLE);
+			return NRF_802154_SL_LPTIMER_PLATFORM_NO_RESOURCES;
+		}
+	}
+
+	if (z_nrf_rtc_timer_set(m_hw_task.chan, fire_lpticks, NULL, NULL) != 0) {
+		hw_task_state_set(HW_TASK_STATE_SETTING_UP, HW_TASK_STATE_IDLE);
+		return NRF_802154_SL_LPTIMER_PLATFORM_TOO_DISTANT;
+	}
+
+	if (ppi_channel != NRF_802154_SL_HW_TASK_PPI_INVALID) {
+		cc_bind_to_ppi(m_hw_task.chan, ppi_channel);
+	}
+	m_hw_task.ppi = ppi_channel;
+
+	nrf_802154_sl_mcu_critical_enter(mcu_cs_state);
+
+	/* For triggering to take place a safe margin is 2 lpticks from `now`. */
+	if ((z_nrf_rtc_timer_read() + 2) > fire_lpticks) {
+		/* it is too late */
+		cc_unbind(m_hw_task.chan, ppi_channel);
+		m_hw_task.ppi = NRF_802154_SL_HW_TASK_PPI_INVALID;
+		z_nrf_rtc_timer_abort(m_hw_task.chan);
+		done_on_time = false;
+	} else {
+		done_on_time = true;
+	}
+
+	nrf_802154_sl_mcu_critical_exit(mcu_cs_state);
+
+	hw_task_state_set(HW_TASK_STATE_SETTING_UP,
+			  done_on_time ? HW_TASK_STATE_READY : HW_TASK_STATE_IDLE);
+
+	return done_on_time ? NRF_802154_SL_LPTIMER_PLATFORM_SUCCESS :
+	       NRF_802154_SL_LPTIMER_PLATFORM_TOO_LATE;
+}
+
+void nrf_802154_platform_sl_lptimer_hw_task_cleanup(void)
```

Comment:
This function doesn't return any result, so it may silently fail. Consider what happens if `cleanup` preempts `update_ppi` - it will take no effect as `m_hw_task.state` will already be set to `HW_TASK_STATE_UPDATING`. Do you think it would make sense to return `nrf_802154_sl_lptimer_platform_result_t` that the API user could use to detect failed cleanups?

Reply:
Sure, returning a result can be useful in case of cleaning problems (I haven't noticed such yet, stability tests are successful, but I'll check it again, also for the `cleanup` and `update_ppi` scenario you mentioned).
I added returning the result

Reply:
In nrf802154 I added asserts on unsuccessful cleanup. Stability tests showed no problems here.